opencga-analysis/src/main/java/org/opencb/opencga/analysis/customTool/CustomToolExecutor.java

```diff
@@ -36,44 +49,96 @@ protected void check() throws Exception {
 
         // Check any condition
         if (runParams == null) {
-            throw new ToolException("Missing runParams");
+            throw new ToolException("Missing CustomExternalToolParams object.");
+        } else if (StringUtils.isEmpty(runParams.getId())) {
+            throw new ToolException("Missing external tool id.");
+        }
+        Container container = generateDockerObject();
+        checkDockerObject(container);
+    }
+
+    private Container generateDockerObject() throws Exception {
+        OpenCGAResult<ExternalTool> result;
+        if (runParams.getVersion() != null) {
+            Query query = new Query(ExternalToolDBAdaptor.QueryParams.VERSION.key(), runParams.getVersion());
+            result = catalogManager.getExternalToolManager().get(study, Collections.singletonList(runParams.getId()), query,
+                    QueryOptions.empty(), false, token);
+        } else {
+            result = catalogManager.getExternalToolManager().get(study, runParams.getId(), QueryOptions.empty(), token);
         }
-        if (StringUtils.isEmpty(runParams.getCommandLine())) {
-            throw new ToolException("Missing commandLine");
+        if (result.getNumResults() == 0) {
+            throw new ToolException("Custom tool '" + runParams.getId() + "' not found");
         }
-        if (runParams.getDocker() == null || StringUtils.isEmpty(runParams.getDocker().getId())) {
-            runParams.setDocker(new JobRunDockerParams("opencb/opencga-ext-tools", "3.2.1", null));
+        ExternalTool externalTool = result.first();
+
+        if (externalTool == null) {
+            throw new ToolException("Custom tool '" + runParams.getId() + "' is null");
         }
-        if (!runParams.getDocker().getId().contains("/")) {
-            throw new ToolException("Missing repository organization. Format for the docker image should be 'organization/image'");
+        if (externalTool.getType() != ExternalToolType.CUSTOM_TOOL) {
+            throw new ToolException("User tool '" + runParams.getId() + "' is not of type " + ExternalToolType.CUSTOM_TOOL);
         }
-        this.dockerImage = runParams.getDocker().getId();
-        if (StringUtils.isNotEmpty(runParams.getDocker().getTag())) {
-            this.dockerImage += ":" + runParams.getDocker().getTag();
+        if (externalTool.getContainer() == null) {
+            throw new ToolException("User tool '" + runParams.getId() + "' does not have a docker object");
```

Comment:
Is this possible? Or is "container" mandatory on this type of tools?
(Not saying that you should remove this check, it just makes me suspicious..)